pkg/virt-launcher/notify-client/client.go

```diff
@@ -117,7 +117,8 @@ func libvirtEventCallback(c cli.Connection, domain *api.Domain, libvirtEvent lib
 
 		spec, err := util.GetDomainSpecWithRuntimeInfo(status, d)
 		if err != nil {
-			if !domainerrors.IsNotFound(err) {
+			// NOTE: Getting domain metadata or a live-migrating VM isn't allowed
+			if !domainerrors.IsNotFound(err) && !domainerrors.IsInvalidOperation(err) {
```

Comment:
Ok, so I personally find that we should switch direction here. Fetching metadata has caused us a lot of trouble in various situations. It is still a relikt from the time where we had a centralized libvirt. We can simply store our metadata in memory in virt-launcher and not touch the metadata at all anymore. @vladikr what do you think?

Reply:
@rmohr  You're right. It seems like the best option. Not sure how to sync it with the virt-handler.